database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -0,0 +1,438 @@
+            createTableSQL.append(fieldName).append(" ").append(fieldType).append(", ");
+        }
+
+        createTableSQL.setLength(createTableSQL.length() - 2);
+        createTableSQL.append(")");
+
+        LOG.info("Executing SQL: {}", createTableSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement()) {
+            statement.executeUpdate(createTableSQL.toString());
+            return true;
+        } catch (SQLException e) {
+            LOG.error(UNABLE_CREATE_TABLE + ": {}", tableName);
+            throw new CreationDatabaseException(UNABLE_CREATE_TABLE + ": " + e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean deleteTable(Class<? extends BaseEntity> entityClass) {
+        String tableName = entityClass.getSimpleName();
+        String dropTableSQL = "DROP TABLE IF EXISTS " + tableName;
+
+        LOG.info("Executing SQL: {}", dropTableSQL);
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement()) {
+            statement.executeUpdate(dropTableSQL);
+            return true;
+        } catch (SQLException e) {
+            LOG.error(UNABLE_DELETE_TABLE + ": {}", tableName);
+            throw new DeletionDatabaseException(UNABLE_DELETE_TABLE);
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity> T addNewRecordToTable(T entity) {
+        if (entity.getId() != null) {
+            throw new IdProvidedManuallyException(ID_PROVIDED_MANUALLY);
+        }
+
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        String tableName = entityClass.getSimpleName();
+
+        String checkTableSQL = "SELECT COUNT(*) FROM information_schema.tables " +
+                "WHERE table_schema = 'entities' AND table_name = '" + tableName + "'";
+
+        String checkEmptyTableSQL = "SELECT COUNT(*) FROM " + tableName;
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(checkTableSQL)) {
+
+            if (resultSet.next() && resultSet.getInt(1) > 0) {
+                try (ResultSet emptyTableResultSet = statement.executeQuery(checkEmptyTableSQL)) {
+                    int rowCount = 0;
+                    if (emptyTableResultSet.next()) {
+                        rowCount = emptyTableResultSet.getInt(1);
+                    }
+
+                    if (rowCount == 0) {
+                        String resetAutoIncrementSQL = "ALTER TABLE " + tableName + " AUTO_INCREMENT = 1";
+                        statement.executeUpdate(resetAutoIncrementSQL);
+                    }
+                }
+
+                String insertSQL = generateInsertSQL(entity, tableName);
```

Comment:
addNewRecordToTable methods looks too big. In the first half, we validate table and prepare incrementor (if needed), then we generate query and execute it. Let's move first half to a separate private method to improve readability.

Reply:
Please log main sql queries like in `createTable` and `deleteTable` here and in all other methods.